bgpd/bgpd.c

```diff
@@ -2279,9 +2279,9 @@ int peer_delete(struct peer *peer)
 	/* Password configuration */
 	if (CHECK_FLAG(peer->flags, PEER_FLAG_PASSWORD)) {
 		XFREE(MTYPE_PEER_PASSWORD, peer->password);
-
 		if (!accept_peer && !BGP_PEER_SU_UNSPEC(peer)
-		    && !CHECK_FLAG(peer->sflags, PEER_STATUS_GROUP))
+		    && !CHECK_FLAG(peer->sflags, PEER_STATUS_GROUP)
+		    && !CHECK_FLAG(peer->flags, PEER_FLAG_DYNAMIC_NEIGHBOR))
```

Comment:
just reading loudly (not sure if I understood correctly): dynamic peers are configured with a peer-group (mandatory), hence isn't this a dead test?

Reply:
They do get configured using a peer group but from my debugging the dynamic peers created from the peer group are not flagged a peer groups themselves.

Reply:
I looked a little deeper into this and it is as I said the dynamic peer gets created in peer_create_bind_dynamic_neighbor it takes some config from the peer group but only has the PEER_FLAG_DYNAMIC_NEIGHBOR and PEER_FLAG_CONFIG_NODE flags set on it.

Reply:
Then maybe a flag is missing to set when binding dynamic to neighbor?

Reply:
It seems dynamic peers should be flagged as peer group members ... at least I would think so

Reply:
I don't think the PEER_STATUS_GROUP flag indicates a group member, rather it identifies the peer-group conf peer structure.